tests/test_text/test_tsne.py

```diff
@@ -45,26 +51,98 @@ def test_make_pipeline(self):
+        Check tSNE integrated visualization on the hobbies corpus
         """
         corpus = self.load_data('hobbies')
         tfidf  = TfidfVectorizer()
 
         docs   = tfidf.fit_transform(corpus.data)
         labels = corpus.target
 
-        tsne = TSNEVisualizer()
+        tsne = TSNEVisualizer(random_state=8392, colormap='Set1')
         tsne.fit_transform(docs, labels)
+        
+        tol = 40 if six.PY3 else 55
+        self.assert_images_similar(tsne, tol=tol)
+
+    def test_make_classification_tsne(self):
+        """
+        Test tSNE integrated visualization on a sklearn classifier dataset
+        """
+
+        ## produce random data
+        X, y = make_classification(n_samples=200, n_features=100,
+                               n_informative=20, n_redundant=10,
+                               n_classes=3, random_state=42)
+
+        ## visualize data with t-SNE
+        tsne = TSNEVisualizer(random_state=87)
+        tsne.fit(X, y)
+    
+        tol = 0.1 if six.PY3 else 40
+        self.assert_images_similar(tsne, tol=tol)
+
+    def test_make_classification_tsne_class_labels(self):
+        """
+        Test tSNE integrated visualization with class labels specified
+        """
+
+        ## produce random data
+        X, y = make_classification(n_samples=200, n_features=100,
+                               n_informative=20, n_redundant=10,
+                               n_classes=3, random_state=42)
+
+        ## visualize data with t-SNE
+        tsne = TSNEVisualizer(random_state=87, labels=['a', 'b', 'c'])
+        tsne.fit(X, y)
+    
+        tol = 0.1 if six.PY3 else 40
+        self.assert_images_similar(tsne, tol=tol)
+
+    def test_tsne_mismtached_labels(self):
+        """
+        Assert exception is raised when number of labels doesn't match
+        """
+        ## produce random data
+        X, y = make_classification(n_samples=200, n_features=100,
+                               n_informative=20, n_redundant=10,
+                               n_classes=3, random_state=42)
+
+        ## fewer labels than classes
+        tsne = TSNEVisualizer(random_state=87, labels=['a', 'b'])
+        with pytest.raises(YellowbrickValueError):
+            tsne.fit(X,y)
+
+        ## more labels than classes
+        tsne = TSNEVisualizer(random_state=87, labels=['a', 'b', 'c', 'd'])
+        with pytest.raises(YellowbrickValueError):
+            tsne.fit(X,y)
+
+
+    def test_no_target_tsne(self):
+        """
+        Test tSNE when no target or classes are specified
+        """
+        ## produce random data
+        X, y = make_classification(n_samples=200, n_features=100,
+                               n_informative=20, n_redundant=10,
+                               n_classes=3, random_state=6897)
+
+        ## visualize data with t-SNE
+        tsne = TSNEVisualizer(random_state=64)
+        tsne.fit(X)
+
+        self.assert_images_similar(tsne, tol=0.1)
```

Comment:
Any chance that we can add a test for handling of pandas dataframes? We promise support in the docs for them, and probably have tests to support it's use.

Reply:
Will do